### DISCUSSION TITLE
OAI 5G Core (SA) Support

### DISCUSSION THREAD

**Discussion by ryanwwest** [srsran/srsRAN_Project#66] [General Help]
I've been trying to run the srsRAN gNB with OAI's 5G Core network. I haven't had luck yet as it fails when trying to set up a Quectel COTS UE's PDU session with the UPF (from the oai-amf log, not srsRAN):

```
192.168.1.2 in the srsRAN gnb conf.
[2023-04-29T04:52:29.341338] [AMF] [ngap   ] [debug] Handle PDU Session Resource Setup Response
[LibNGAP]Received RanUeNgapId 1
[2023-04-29T04:52:29.341360] [AMF] [ngap   ] [error] Decoding PduSessionResourceSetupResponseMsg getPduSessionResourceSetupResponseList IE error
Decoded successfully
rc.consumed to decode = 2
```

Everything seems to work fine when the gNB initially associates with the AMF. However, I know that Open5GS is the supported and recommended core. I was wondering if OAI's core should also be supported, if it has been tested, or if it's known that it won't work yet and that I need to use Open5GS instead? Additionally, I'd be curious as to why Open5GS is preferred over OAI here (e.g., is is more mature or reliable?).

**Comment by ryanwwest** [comment]
(Expanded message from oai-amf just before the above error logs:

```
[2023-04-29T05:00:01.621419] [AMF] [ngap   ] [debug] Decoded NGAP message, procedure code 29, present 2
    value: PDUSessionResourceSetupResponse ::= {
        protocolIEs: ProtocolIE-Container ::= {
            PDUSessionResourceSetupResponseIEs ::= {
                id: 10
                criticality: 1 (ignore)
                value: 1
            }
            PDUSessionResourceSetupResponseIEs ::= {
                id: 85
                criticality: 1 (ignore)
                value: 1
            }
            PDUSessionResourceSetupResponseIEs ::= {
                id: 58
                criticality: 1 (ignore)
                value: PDUSessionResourceFailedToSetupListSURes ::= {
                    PDUSessionResourceFailedToSetupItemSURes ::= {
                        pDUSessionID: 1
                        pDUSessionResourceSetupUnsuccessfulTransfer: 0C 00
                    }
                }
            }
        }
    }
}
```
)

**Comment by andrepuschmann** [comment]
Hey,
we know of people having used OAI core successfully but I haven't tried myself. Could you please share a full NGAP PCAP and gNB logs so we can have a look?

**Comment by andrepuschmann** [comment]
Closing the discussion due to missing feedback from original author.
